lambda/jvm/src/main/scala/feral/lambda/IOLambdaPlatform.scala

```diff
@@ -29,11 +32,33 @@ import java.io.OutputStream
 import java.io.OutputStreamWriter
 import java.nio.channels.Channels
 import scala.concurrent.duration._
+import scala.util.control.NonFatal
 
 private[lambda] abstract class IOLambdaPlatform[Event, Result]
     extends lambdaRuntime.RequestStreamHandler { this: IOLambda[Event, Result] =>
 
   private[this] val (dispatcher, handle) = {
+    val handler = {
+      val h =
+        try this.handler
+        catch { case ex if NonFatal(ex) => null }
+
+      if (h ne null) { h.map(IO.pure(_)) }
+      else {
+        val lambdaName = getClass().getSimpleName()
+        val msg =
+          s"""|There was an error initializing `$lambdaName` during startup.
+              |Falling back to initialize-during-first-invocation strategy.
+              |To fix, try replacing any `val`s in `$lambdaName` with `def`s.""".stripMargin
+        System.err.println(msg)
+
+        Async[Resource[IO, *]].defer(this.handler).memoize.map {
+          case Resource.Eval(ioa) => ioa
+          case _ => throw new AssertionError
+        }
```

Comment:
There will be an official way to do this soon.
- https://github.com/typelevel/cats-effect/pull/3890